devicetypes/qubino/qubino-flush-shutter.src/qubino-flush-shutter.groovy

```diff
@@ -0,0 +1,501 @@
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	if (encapsulatedCommand) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "unable to extract secure command from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd, ep = null) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	zwaveEvent(encapsulatedCommand, cmd.sourceEndPoint as Integer)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelReport cmd, ep = null) {
+	log.debug "SwitchMultilevelReport ${cmd} from endpoint: ${ep}"
+	if (cmd.value != 0xFE) {
+		if (ep != 2) {
+			shadeEvent(cmd.value)
+		} else {
+			def event = [name: "level", value: cmd.value != 0x63 ? cmd.value : 100]
+			sendEventsToVenetianBlind([event])
+		}
+	} else {
+		log.warn "Something went wrong with calibration, position of blind is unknown"
+		if (ep == 2) {
+			sendEventsToVenetianBlind([[name: "level", value: 0]])
+		} else {
+			[
+					createEvent([name: "windowShade", value: "unknown"]),
+					createEvent([name: "level", value: 0]),
+					createEvent([name: "shadeLevel", value: 0])
+			]
+		}
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchmultilevelv3.SwitchMultilevelSet cmd, ep = null) {
+	sendHubCommand(encap(zwave.meterV3.meterGet(scale: 0x02)))
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd, ep = null) {
+	log.debug "BasicReport ${cmd}"
+	if (cmd.value != 0xFE && ep != 2) {
+		shadeEvent(cmd.value)
+	} else {
+		log.warn "Something went wrong with calibration, position of blind is unknown"
+	}
+}
+
+private shadeEvent(value) {
+	def shadeValue
+	def events = []
+	if (!value) {
+		shadeValue = "open"
+	} else if (value == 0x63) {
+		shadeValue = "closed"
+	} else {
+		shadeValue = "partially open"
+	}
+	events += createEvent([name: "windowShade", value: shadeValue])
+	events += createEvent([name: "level", value: value != 0x63 ? value : 100])
+	events += createEvent([name: "shadeLevel", value: value != 0x63 ? value : 100])
+
+	events
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd, ep = null) {
+	def events = []
+	if (cmd.meterType == 0x01) {
+		def eventMap = [:]
+		if (cmd.scale == 0x00) {
+			eventMap.name = "energy"
+			eventMap.value = cmd.scaledMeterValue
+			eventMap.unit = "kWh"
+			events += createEvent(eventMap)
+		} else if (cmd.scale == 0x02) {
+			eventMap.name = "power"
+			eventMap.value = Math.round(cmd.scaledMeterValue)
+			eventMap.unit = "W"
+			events += createEvent(eventMap)
+			if (cmd.scaledMeterValue) {
+				events += createEvent([name: "windowShade", value: state.blindsLastCommand])
+				events += [createEvent([name: "level", value: state.shadeTarget, displayed: false]), createEvent([name: "shadeLevel", value: state.shadeTarget, displayed: false])]
+			} else {
```

Comment:
why do you do a get here if the meter value is 0?

Reply:
Blinds movement can be triggered manually with switches, so there's no useful value in `state.shadeTarget`. I'm using get to show actual position when motor stops.

Reply:
In those cases it doesn't send an unsolicited multilevel report, but does send an empty meter report?